packages/styletron-engine-atomic/src/server/server.js

```diff
@@ -117,12 +130,74 @@ class StyletronServer implements StandardEngine {
             },
           ]
         : []),
-      ...sheetify(this.styleRules, this.styleCache.getSortedCacheKeys()),
+      ...sheetify(
+        this.styleRules,
+        this.styleCache.getSortedCacheKeys(),
+        this.streamingMode,
+      ),
     ];
+
+    // Clear streamed styles
+    if (this.streamingMode) {
+      this.keyframesRules = "";
+      this.fontFaceRules = "";
+      for (const prop in this.styleRules) {
+        this.styleRules[prop].rules = "";
+      }
+    }
+
+    return sheets;
   }
 
   getStylesheetsHtml(className?: string = "_styletron_hydrate_") {
-    return generateHtmlString(this.getStylesheets(), className);
+    return generateHtmlString(
+      this.getStylesheets(),
+      className,
+      this.streamingMode,
+    );
+  }
+
+  // Copied and adapted from: https://github.com/styled-components/styled-components/blob/28b4c28f1e/packages/styled-components/src/models/ServerStyleSheet.js#L78
+  getStylesheetsStream(htmlStream: Readable) {
+    if (!this.streamingMode)
+      throw new Error(
+        "getStylesheetsStream() requires streamingMode to be set to true for Styletron server",
+      );
+
+    const readableStream: Readable = htmlStream;
+    const CLOSING_TAG_R = /^\s*<\/[a-z]/i;
+
+    const instance = this;
+
+    const transformer = new Transform({
+      transform: function appendStyleChunks(chunk, /* encoding */ _, callback) {
+        // Get the chunk and retrieve the sheet's CSS as an HTML chunk,
+        // then reset its rules so we get only new ones for the next chunk
```

Comment:
Which piece actually does the reset for the next chunk?

Reply:
This is where that's handled:

https://github.com/AyoubElk/styletron/blob/01edeb3dd4031d963cd0c1b69520663e2fd377a9/packages/styletron-engine-atomic/src/server/server.js#L140-L147